pkg/controller/sparkapplication/controller.go

```diff
@@ -442,7 +442,9 @@ func (c *Controller) getAndUpdateAppState(app *v1beta2.SparkApplication) error {
 }
 
 func (c *Controller) handleSparkApplicationDeletion(app *v1beta2.SparkApplication) {
-	c.metrics.exportMetricsOnDelete(app)
+	if c.metrics != nil {
```

Comment:
manually highlight changes that I added. this fixes the nil pointer on SparkApp deletion